opendut-util/src/logging/mod.rs

```diff
@@ -229,6 +247,51 @@ pub enum OpenTelemetryConfig {
     #[default]
     Disabled,
 }
+
+#[derive(Clone)]
+struct MyArcMutex<T>(Arc<Mutex<T>>);
+
+impl Interceptor for MyArcMutex<Option<ConfidentialClientRef>> {
```

Comment:
This seems to have no dependencies on opentelemetry and can be moved to opendut_auth crate: opendut_auth::telemetry::mod.rs